packages/babel-parser/src/plugins/typescript/index.ts

```diff
@@ -447,6 +447,10 @@ export default (superClass: ClassWithMixin<typeof Parser, IJSXParserMixin>) =>
             modifier,
           });
         }
+
+        if (this.hasPrecedingLineBreak()) {
```

Comment:
We already check the newline in `tsNextTokenCanFollowModifier`, but somehow we missed this check for the `static` token, could you move the newline check there? So we don't have to check the line break for every other modifiers here.